gems/smithy/lib/smithy/plan.rb

```diff
@@ -3,22 +3,35 @@
 module Smithy
   # The Plan class is a simple data structure that holds the model, type, and options for a generator.
   class Plan
-    # @param model [Hash] The API model as a JSON hash.
-    # @param type [Symbol] The type of code to generate, either :client, :server, or :types.
-    # @param options [Hash] The options passed to the generator.
+    # @param [Hash] model The API model as a JSON hash.
+    # @param [Symbol] type The type of code to generate, either :client, :server, or :types.
+    # @param [Hash] options The options passed to the generator.
     def initialize(model, type, options = {})
-      @model = Vise::Model.new(model)
       @type = type
       @options = options
+
+      Welds.load!(self)
+      @welds = Welds.for(model)
+      Polishes.load!(self)
+      @polishes = Polishes.for(model)
+
+      @welds.each { |weld| weld.preprocess(model) }
+      @model = Vise::Model.new(model)
     end
 
-    # @return [Hash] The API model as a JSON hash.
+    # @return [Vise::Model] The API model wrapped by Vise.
     attr_reader :model
 
     # @return [Symbol] The type of code to generate.
     attr_reader :type
 
     # @return [Hash] The options passed to the generator.
     attr_reader :options
+
+    # @return [Array<Weld>] The welds that apply to this model.
```

Comment:
nit: apply to this plan.  (since in theory they may apply only for client or server, ect).